examples/on-conflict-clause/on_conflict_clause.bal

```diff
@@ -1,38 +1,61 @@
 import ballerina/io;
 
-type Student record {|
+type Employee record {|
     readonly int id;
     string name;
+    string role;
     int score;
 |};
 
 public function main() {
-    error onConflictError = error("Key Conflict", message = "record with same key exists.");
-
-    table<Student> students = table [
-            {id: 1, name: "John", score: 100},
-            {id: 2, name: "Jane", score: 150},
-            {id: 1, name: "John", score: 200}
+    table<Employee> employees = table [
+            {id: 1, name: "Mike", role: "Admin", score: 100},
+            {id: 2, name: "John", role: "HR", score: 150},
+            {id: 3, name: "Amy", role: "HR", score: 150},
+            {id: 1, name: "Mike", role: "Admin", score: 200}
         ];
 
-    // The result of the following will be an error since the key `John` is duplicated.
-    map<int>|error studentScores = map from var {name, score} in students
+    // The result of the following will be an error since the key `Mike` is duplicated
+    // and the `replaceIfSafe()` function returns an error for the role `Admin`.
+    map<int>|error employeeScores = map from var {name, score, role} in employees
                                    select [name, score]
-                                   on conflict onConflictError;
+                                   on conflict replaceIfSafe(role);
+
+    io:println(employeeScores);
 
-    io:println(studentScores);
+    // The result of the following will be an error since the key `1` is duplicated
+    // and `replaceIfSafe()` function returns an error for the role `Admin`.
+    table<Employee> key(id)|error employeeScoresTable = table key(id) from var employee in employees
+                                                      select employee
+                                                      on conflict replaceIfSafe(employee.role);
```

Comment:
Here again, it's a bit confusing to replace by checking role when the id is where the conflict is, IMO.

Reply:
That is the idea right if there is a conflicting key we make decisions using the frame information. Not only the key